cylc/flow/etc/cylc

```diff
@@ -96,6 +96,12 @@ CYLC_HOME_ROOT="${CYLC_HOME_ROOT:-/opt}"
 # export ROSE_SITE_CONF_PATH="${ROSE_SITE_CONF_PATH:-/etc/rose}"
 ###############################################################################
 
+# Prevent user's PyhtonPath interfering with Cylc:
+if [[ -n "${PYTHONPATH}" ]]; then
```

Comment:
It's more complicated than this. We need to know what we are running.
I think we need:
- Cylc 8: unset PYTHONPATH (no need to save but no harm if we do)
- Rose 2: unset PYTHONPATH and save in _PYTHONPATH
- Cylc 7: do nothing (might be OK to unset but safer not to)
- Rose 2019: do nothing (would break Rose apps if we unset)

Reply:
Not sure if this is sufficient (moved further down the script):

`if [[ -n "${PYTHONPATH}" && ${CYLC_ENV_NAME:-} =~ ^cylc-8 ]]; then`

Reply:
Thanks @dpmatthews : The list of conditions is probably what I needed to get further.